### DISCUSSION TITLE
How to Change MenuBar App Icon?

### DISCUSSION THREAD

**Discussion by ZhiaDev** [mynameiskenlee/flutter_macos_menubar_example#14] [General]
Hello,
I have a question regarding my Flutter application. How can I change the default Flutter logo icon in the app's menu bar?

Additionally, is it possible to use a dynamic icon? For instance, I'm developing a VPN app, and I'd like the app icon to change based on whether the VPN is connected or disconnected

Thank you for your help!

**Comment by mynameiskenlee** [comment]
Hi @ZhiaDev,

Since the app icon is not controlled by flutter, you have to put your app icon into the assets folder in your Xcode project and update the `statusBarButton.image` at [macos/Runner/StatusBarController.swift](https://github.com/mynameiskenlee/flutter_macos_menubar_example/blob/a056ea3f4f727e46f6c5f94996cfe2ac0c3f2de7/macos/Runner/StatusBarController.swift#L21C66-L21C73)

For dynamic icon, I think that is possible through [Flutter Platform Channels](https://docs.flutter.dev/platform-integration/platform-channels) and update the image programmatically. 

Feel free to let me know if you have any other issue :)

**Comment by ZhiaDev** [comment]
Hey @mynameiskenlee,

I just wanted to drop you a quick note to say thanks a ton for that awesome repository you put together. It really helped me out a bunch. With your repository and the tips you gave me for changing the program icon, I managed to whip up a cool [proxy management app](https://github.com/ZhiaDev/mac-proxy-manager) for the Mac MenuBar.

So, thanks again for your help, and sorry for the late response. Much appreciated!

Cheers, Zhia
